app/controllers/attendances_controller.rb

```diff
@@ -0,0 +1,23 @@
+class AttendancesController < ApplicationController
+  def new
+    @attendance = Attendance.new
+  end
+
+  def create
+    @attendance = Attendance.new(attendance_params)
+    
+    if @attendance.save
+      flash[:success] = "You attended this event!"
+      redirect_to root_path
+    else
+      flash[:warning] = "the event has already passed"
```

Comment:
guys redirect creates a new request let's save that trip and do this:
`flash.now[:warning] = "the event has already passed"
render :event_path.`
this way we get back with out flash too